tests/kubevirt_helper.go

```diff
@@ -477,24 +485,163 @@ func AreVolumeReplicasCollocated(vol *volume.Volume, globalReplicSet []*api.Repl
+	}
+	return nil
+}
+
+// ListEvents lists all events in a namespace in logs.
+func ListEvents(namespace string) error {
+	eventList, err := k8sCore.ListEvents(namespace, metav1.ListOptions{})
+	if err != nil {
+		log.Infof("Failed to list events in namespace %s: %v", namespace, err)
+		return err
+	}
+	log.Infof("Events in namespace %s:", namespace)
+	for _, event := range eventList.Items {
+		log.Infof("Time: %v, Event: %s, Type: %s, Reason: %s, Object: %s/%s, Message: %s",
+			event.FirstTimestamp, event.Name, event.Type, event.Reason, event.InvolvedObject.Kind, event.InvolvedObject.Name, event.Message)
+	}
+	return nil
+}
+
+// HotAddPVCsToKubevirtVM hot adds disk to a running VM
+func HotAddPVCsToKubevirtVM(virtualMachines []*scheduler.Context, numberOfDisks int, size string) error {
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
 		if err != nil {
-			log.Fatalf("failed to create config map kubevirt-creds: %v", err)
+			return fmt.Errorf("failed to get VMs from context: %w", err)
+		}
+		for _, v := range vms {
+			diskCountOutput, err := GetNumberOfDisksInVM(v)
+			log.Infof("Currently having %v number of disks in the VM", diskCountOutput)
+			storageClass, err := GetStorageClassOfVmPVC(appCtx)
+			if err != nil {
+				return fmt.Errorf("failed to get storage class for VM [%s]: %w", v.Name, err)
+			}
+			pvcs, err := CreatePVCsForVM(v, numberOfDisks, storageClass, size)
+			if err != nil {
+				return fmt.Errorf("failed to create PVCs for VM [%s]: %w", v.Name, err)
+			}
+
+			vm, err := k8sKubevirt.GetVirtualMachine(v.Name, v.Namespace)
+			if err != nil {
+				return fmt.Errorf("failed to get VM [%s]: %w", v.Name, err)
+			}
+
+			for _, pvc := range pvcs {
+				diskName := fmt.Sprintf("disk-%s", pvc.Name)
+				// Appending the disk and volume definitions to the VM spec
+				vm.Spec.Template.Spec.Domain.Devices.Disks = append(vm.Spec.Template.Spec.Domain.Devices.Disks, kubevirtv1.Disk{
+					Name: diskName,
+					DiskDevice: kubevirtv1.DiskDevice{
+						Disk: &kubevirtv1.DiskTarget{
+							Bus: "virtio",
+						},
+					},
+				})
+
+				vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, kubevirtv1.Volume{
+					Name: diskName,
+					VolumeSource: kubevirtv1.VolumeSource{
+						PersistentVolumeClaim: &kubevirtv1.PersistentVolumeClaimVolumeSource{
+							PersistentVolumeClaimVolumeSource: corev1.PersistentVolumeClaimVolumeSource{
+								ClaimName: pvc.Name,
+							},
+							Hotpluggable: true,
+						},
+					},
+				})
+			}
+
+			_, err = k8sKubevirt.UpdateVirtualMachine(vm)
+			if err != nil {
+				return fmt.Errorf("failed to update VM [%s]: %w", vm.Name, err)
+			}
+
+			err = RestartKubevirtVM(v.Name, v.Namespace, true)
+			if err != nil {
+				return err
+			}
+			log.InfoD("Sleep for 5mins for vm to come up")
+			time.Sleep(5 * time.Minute)
+			NewDiskCountOutput, err := GetNumberOfDisksInVM(v)
```

Comment:
How about we handle error?

Reply:
Good catch - I missed it. Let me do that